File: LeetCode/cpp/1054.cpp
class Solution {
public:
    vector<int> rearrangeBarcodes(vector<int> &barcodes) {
        int i = 0, n = barcodes.size();
        unordered_map<int, int> count;
        vector<int> res(n);
        vector<vector<int>> match;
        for (auto b:barcodes)
            ++count[b];
        for (auto &c:count)
            match.push_back(vector<int>{c.first, c.second});
        sort(match.begin(), match.end(), [](vector<int> &m1, vector<int> &m2) {
            return m1[1] > m2[1];
        });
        for (auto &m:match) {
            for (int &j = m[1]; j > 0; --j) {
                res[i] = m[0];
                i += 2;
                if (i >= n)
                    i = 1;
            }
        }
        return res;
    }
};

File: LeetCode/python/1054.py
class Solution(object):
    def rearrangeBarcodes(self, codes):
        """
        :type barcodes: List[int]
        :rtype: List[int]
        """
        n = len(codes)
        if n <= 2:
            return codes
        for i in range(n):
            j = i + 1
            while j < n and codes[j] == codes[i]:
                j += 1
            if j >= n:
                break
            codes[j], codes[i + 1] = codes[i + 1], codes[j]
        for i in range(n - 1, -1, -1):
            j = i - 1
            while j > -1 and codes[j] == codes[i]:
                j -= 1
            if j < 0:
                break
            codes[j], codes[i - 1] = codes[i - 1], codes[j]
        return codes
